### PR TITLE
feat: add reorder column demo (angular)

### PR DESCRIPTION
**Pull request by steve-haar**
This is a redo of the column ordering for angular demo that adds a11y keyboard support. There is a hack in this that accesses private properties of the datagrid (keyboradNavigationController). However, if we like this approach, we could always open up the API allowing us to remove this hack.

### CODE REVIEW

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
+  }
+
+  reorderColumn(event: { previousIndex: number; currentIndex: number }) {
+    if (event.currentIndex !== event.previousIndex) {
+      const value = [...this.columns];
+      moveItemInArray(value, event.previousIndex, event.currentIndex);
```
Comment (ashleyryan) [inline grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts:36]: I noticed when I used this function in my implementation, that if I was moving the column to the right (to later in the array) I needed to subtract 1 from the target index since everything after the previousIndex will shift left. I haven't pulled down your branch, but you may have the same issue

File: grid-demo/src/app/angular/column-ordering/column-ordering.component.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { CdkDragDrop } from '@angular/cdk/drag-drop';
+import { moveItemInArray } from '@angular/cdk/drag-drop';
```
Comment (ashleyryan) [inline grid-demo/src/app/angular/column-ordering/column-ordering.component.ts:1]: this isn't used in this file

Comment (steve-haar) [reply grid-demo/src/app/angular/column-ordering/column-ordering.component.ts:1]: Thanks, fixed

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts (hunk 1 above)
Comment (steve-haar) [reply grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts:36]: My old implementation was indeed broken. I believe this one works as expected. I've tested it and it seems to work.

**Review by kevinbuhmann — APPROVED** [review]
Approved, but I'd reword the commit message to "feat: rework and improve angular reorder column demo" or similar for accuracy/completeness.

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts (hunk 1 above)
Comment (steve-haar) [reply grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-grid.directive.ts:36]: Looks like the Angular cdk `dropped` event handles this case for us, unlike the `cdsDraggableChange` event.
